clustermesh/clustermesh.go

```diff
@@ -445,6 +445,7 @@ type k8sClusterMeshImplementation interface {
 	ListPods(ctx context.Context, namespace string, options metav1.ListOptions) (*corev1.PodList, error)
 	AutodetectFlavor(ctx context.Context) k8s.Flavor
 	CiliumStatus(ctx context.Context, namespace, pod string) (*models.StatusResponse, error)
+	CiliumDbgEndpoints(ctx context.Context, namespace, pod string) ([]*models.Endpoint, error)
```

Comment:
> IMO the name CiliumDbgEndpoints isn't ideal as there's no context what the dbg stands for 

Yeah, I agree it is suboptimal. I couldn't find a much better name honestly though, to make it distinguishable from `ListCiliumEndpoints` which retrieves the CRD resources. One of my initial attempts was `CiliumStatusEndpoints`, but that doesn't look great either (especially because it is not status specific).

> In general i don't get why the "k8s client" also serves the purpose of providing information that are gathered via cilium-dbg

Totally agree. I also don't get the point of these huge interfaces with the different methods just aliasing the k8s client functionalities. Even for unit testing (which we don't do that much for the CLI in any case) it is just way easier to use a fake client compared to implementing a mock of these interfaces. But that's how it is right now, and any changes seem out of scope (especially because `CiliumStatus` is already part of it).